File: app/src/main/java/com/example/schedule/logic/ScheduleHelper.java
package com.example.schedule.logic;

import com.example.schedule.model.Schedule;
import com.google.gson.Gson;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Calendar;

public class ScheduleHelper {

    private static Schedule mainSchedule = null;

    public static Schedule getSchedule() { // false: short; true: full
        return mainSchedule;
    }

    public static void setSchedule(String json) {
        mainSchedule = new Gson().fromJson(json, Schedule.class);
    }

    public static void loadSchedule(FileInputStream inputStream) throws IOException {
        StringBuilder json = new StringBuilder();

        BufferedReader br = new BufferedReader(new InputStreamReader(inputStream));
        String s;
        while ((s = br.readLine()) != null) json.append(s);
        br.close();

        if (json.toString().isEmpty()) throw new IOException("Пустой файл с расписанием");

        setSchedule(json.toString());
    }

    public static void saveSchedule(String json, FileOutputStream outputStream) throws IOException {
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(outputStream));
        bw.write(json);
        bw.close();
    }

    public static boolean isEven(Calendar date) { // true - Нижняя, false - Верхняя
        return (date.get(Calendar.WEEK_OF_YEAR) % 2) == 0;
    }
}

File: app/src/main/java/com/example/schedule/logic/ScheduleHelper.kt
package com.example.schedule.logic

import android.content.Context
import com.example.schedule.ScheduleApplication
import com.example.schedule.ScheduleApplication.Companion.GROUP_FILE
import com.example.schedule.model.Group
import com.google.gson.Gson
import java.io.*
import java.util.*

object ScheduleHelper {

    var appContext: Context? = null
        set(value) { field = value?.applicationContext }

    private val simpleGson = Gson()

    var group: Group? = null

    fun loadGroup(context: Context): Group? {
        var json: String? = null
        try {
            val br = BufferedReader(InputStreamReader(context.openFileInput(GROUP_FILE)))

            val jsonBuilder = StringBuilder()
            var s: String?
            while (br.readLine().also { s = it } != null) jsonBuilder.append(s)
            br.close()

            json = jsonBuilder.toString()
        } catch (e: IOException) {
            e.printStackTrace()
        }

        //println("Json: ${json == null} $json")

        return if (json != null && json.isNotEmpty()) {
            simpleGson.fromJson(json, Group::class.java)
        } else {
            null
        }
    }

    fun loadGroup(): Boolean {
        val group = loadGroup(appContext!!)
        return if (group != null) {
            this.group = group
            true
        } else false
    }

    private fun setSchedule(json: String) {
        group = simpleGson.fromJson(json, Group::class.java)
    }

    fun saveGroup(group: Group): Boolean {
        return try {
            val jsonGroup: String = simpleGson.toJson(group)
            val bw = BufferedWriter(OutputStreamWriter(appContext!!.openFileOutput(GROUP_FILE, Context.MODE_PRIVATE)))
            bw.write(jsonGroup)
            bw.close()
            true
        } catch (e: IOException) {
            e.printStackTrace()
            false
        }
    }

    fun isEven(date: Calendar): Boolean { // true - Нижняя, false - Верхняя
        val even = date[Calendar.WEEK_OF_YEAR] % 2 == 0
        return if (ScheduleApplication.eveningChanged) !even else even
    }
}
